Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -0,0 +1,151 @@
+        }
+
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let session = try client.startSession()
+        let options = ChangeStreamOptions(fullDocument: .updateLookup)
+        let opts = try encodeOptions(options: options, session: session)
+        let pipeline: Document = []
+
+        try client.connectionPool.withConnection { conn in
+            try db.withMongocDatabase(from: conn) { dbPtr in
+                // TODO: Use MongoDatabase.watch() instead `mongoc_database_watch` of once it gets added
+                let changeStreamPtr: OpaquePointer = mongoc_database_watch(dbPtr, pipeline._bson, opts?._bson)
+                var replyPtr = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
+                defer {
+                    replyPtr.deinitialize(count: 1)
+                    replyPtr.deallocate()
+                }
+
+                let decoder = BSONDecoder()
+                let changeStream = try ChangeStream<ChangeStreamDocument<Document>>(stealing: changeStreamPtr,
+                                                                                    client: client,
+                                                                                    decoder: decoder)
+                // expect the first iteration to be nil since no changes have been made to the database.
+                expect(changeStream.next()).to(beNil())
+
+                let coll = db.collection(self.getCollectionName(suffix: "1"))
+                try coll.insertOne(["a": 1], session: session)
+
+                // test that the change stream contains a change document for the `insert` operation.
+                let res1 = changeStream.next()
+                expect(res1).toNot(beNil())
+                expect(res1?.operationType).to(equal(.insert))
+                expect(res1?.fullDocument?["a"]).to(bsonEqual(1))
+
+                // test that the change stream contains a change document for the `drop` operation.
+                try db.drop()
+                let res2 = changeStream.next()
+                expect(res2).toNot(beNil())
+                expect(res2?.operationType).to(equal(.drop))
+            }
+        }
+    }
+
+    func testChangeStreamOnACollection() throws {
+        guard MongoSwiftTestCase.topologyType != .single else {
+            print("Skipping test case because of unsupported topology type \(MongoSwiftTestCase.topologyType)")
+            return
+        }
+        let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+        let session = try client.startSession()
+        let options = ChangeStreamOptions(fullDocument: .updateLookup)
+        let opts = try encodeOptions(options: options, session: session)
+        let pipeline: Document = []
+
+        try client.connectionPool.withConnection { conn in
```

Comment:
just FYI: this approach is fine for the purpose of this test, but in the next PR when you add watch methods you'll want to switch to doing 
```swift
let conn = try client.connectionPool.checkOut()
try coll.withMongocCollection(from: conn) { collPtr in
    ...
    let changeStream = try ChangeStream<ChangeStreamDocument<Document>>(stealing: changeStreamPtr,
                                                                        client: client,
                                                                        connection: conn, // changestream stores source conn
                                                                        session: session,
                                                                        decoder: decoder)
```

and then `ChangeStream.deinit` will do `self.client.connectionPool.checkIn(self.connection)`.

this will ensure that a `mongoc_client_t` is not destroyed before a `mongoc_change_stream_t` that depends on it is.
(note `mongoc_change_stream_t` stores/uses a pointer its source `mongoc_client_t` - [see here](https://github.com/mongodb/mongo-c-driver/blob/c88b84ff383386f2491ff5298f1a1fd7bae6dd7a/src/libmongoc/src/mongoc/mongoc-change-stream-private.h#L47) - but it's up to us to ensure the client isn't destroyed)

Reply:
Updated. In the future, `MongoCollection` will no longer store a reference to `mongoc_client_t`, so `ChangeStream` would just store a `Connection` instead? I added a todo similar to the one in `MongoClient` : `// TODO: SWIFT-374: remove this property.`
Let me know if this is okay.

Reply:
actually, `MongoCollection` will always need to store a `MongoClient` to ensure that the `MongoClient` and its `ConnectionPool` stay alive for as long as the `MongoCollection` does. So no todo needed.